README.txt
License for all but the tika.jar file: GNU GPL 2.0
   tika.jar is licensed with the ASF License (Apache)

Installation: See, INSTALL.txt

Description:

eZ Tika is an extension that enables a handler for converting multiple binary file formats to plain text as used by the search engine (if you enabled those attributes as searcheable)

Currently, the following formats are enabled (see also binaryfile.ini.append.php for a more up to date list):

[application/pdf]
[application/msword]
[application/vnd.ms-excel]
[application/vnd.ms-powerpoint]
[application/vnd.visio]
[application/vnd.ms-outlook]
[application/xml]
[application/rtf]
[application/vnd.oasis.opendocument.text]
[application/vnd.oasis.opendocument.presentation]
[application/vnd.oasis.opendocument.spreadsheet]
[application/vnd.oasis.opendocument.formula]
[application/zip]
[application/octet-stream]


RELEASES.txt
eztika 1.0 svn revision 3
eztika 1.1 svn revision 15
eztika 1.2 svn revision 22


settings/binaryfile.ini.append.php
<?php /*

[HandlerSettings]

ExtensionRepositories[]=eztika/classes/datatypes/ezbinaryfile

MetaDataExtractor[application/pdf]=ezmulti
MetaDataExtractor[application/msword]=ezmulti
MetaDataExtractor[application/vnd.ms-excel]=ezmulti
MetaDataExtractor[application/vnd.ms-powerpoint]=ezmulti
MetaDataExtractor[application/vnd.visio]=ezmulti
MetaDataExtractor[application/vnd.ms-outlook]=ezmulti
MetaDataExtractor[application/xml]=ezmulti
MetaDataExtractor[application/rtf]=ezmulti
MetaDataExtractor[application/vnd.oasis.opendocument.text]=ezmulti
MetaDataExtractor[application/vnd.oasis.opendocument.presentation]=ezmulti
MetaDataExtractor[application/vnd.oasis.opendocument.spreadsheet]=ezmulti
MetaDataExtractor[application/vnd.oasis.opendocument.formula]=ezmulti
MetaDataExtractor[application/zip]=ezmulti
#Office 2007 word file, but not yet supported in eZP4.0,eZP4.1
MetaDataExtractor[application/vnd.openxmlformats-officedocument.wordprocessingml.document]=ezmulti
#octet-stream is the default, and in eZP4.O Office 2007 files are assigned this mime type
#If tika senses a known format like the Office 2007 types, it will treat them well
MetaDataExtractor[application/octet-stream]=ezmulti
[MultiHandlerSettings]

#Change the path to the eztika shell script, which probably needs editing as well
TextExtractionTool=/opt/local/bin/eztika

 */
?>

settings/binaryfile.ini.append.withxpdf.php
<?php /*

[HandlerSettings]

ExtensionRepositories[]=eztika/classes/datatypes/ezbinaryfile

#warning: CJK pdf's are not working correctly, use xpdf based conversion instead
MetaDataExtractor[application/pdf]=ezpdf

MetaDataExtractor[application/msword]=ezmulti
MetaDataExtractor[application/vnd.ms-excel]=ezmulti
MetaDataExtractor[application/vnd.ms-powerpoint]=ezmulti
MetaDataExtractor[application/vnd.visio]=ezmulti
MetaDataExtractor[application/vnd.ms-outlook]=ezmulti
MetaDataExtractor[application/xml]=ezmulti
MetaDataExtractor[application/rtf]=ezmulti
MetaDataExtractor[application/vnd.oasis.opendocument.text]=ezmulti
MetaDataExtractor[application/vnd.oasis.opendocument.presentation]=ezmulti
MetaDataExtractor[application/vnd.oasis.opendocument.spreadsheet]=ezmulti
MetaDataExtractor[application/vnd.oasis.opendocument.formula]=ezmulti
MetaDataExtractor[application/zip]=ezmulti
#Office OOXML files, supported since eZP4.0.4, eZP4.1.0
MetaDataExtractor[application/vnd.openxmlformats-officedocument.wordprocessingml.document]=ezmulti
MetaDataExtractor[application/vnd.openxmlformats-officedocument.spreadsheetml.sheet]=ezmulti
MetaDataExtractor[application/vnd.openxmlformats-officedocument.presentationml.presentation]=ezmulti
#octet-stream is the default
#If tika senses a known format like the Office 2007 ooxml types, it will treat them well
MetaDataExtractor[application/octet-stream]=ezmulti
[MultiHandlerSettings]

#Change the path to the eztika shell script, which probably needs editing as well
TextExtractionTool=/opt/local/bin/eztika


[PDFHandlerSettings]
TextExtractionTool=/opt/local/bin/ezpdftotext
 */
?>

classes/datatypes/ezbinaryfile/plugins/ezmultiparser.php
<?php
//
// Definition of eZMultiParser class
//
// Created on: <08-Dec-2008 20:13 pb>
//
// ## BEGIN COPYRIGHT, LICENSE AND WARRANTY NOTICE ##
// SOFTWARE NAME: eZ Tika
// SOFTWARE RELEASE: 1.1
// COPYRIGHT NOTICE: Copyright (C) 2008-2009 Paul Borgermans
// SOFTWARE LICENSE: GNU General Public License v2.0
// NOTICE: >
//   This program is free software; you can redistribute it and/or
//   modify it under the terms of version 2.0  of the GNU General
//   Public License as published by the Free Software Foundation.
//
//   This program is distributed in the hope that it will be useful,
//   but WITHOUT ANY WARRANTY; without even the implied warranty of
//   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
//   GNU General Public License for more details.
//
//   You should have received a copy of version 2.0 of the GNU General
//   Public License along with this program; if not, write to the Free
//   Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,
//   MA 02110-1301, USA.
//
//
// ## END COPYRIGHT, LICENSE AND WARRANTY NOTICE ##
//

/*!
  \class eZMultiParser ezmultiparser.php
  \ingroup eZKernel
  \brief The class eZMultiParser handles parsing of multiple files and returns the metadata

*/

class eZMultiParser
{
    function parseFile( $fileName )
    {
        $binaryINI = eZINI::instance( 'binaryfile.ini' );

        $textExtractionTool = $binaryINI->variable( 'MultiHandlerSettings', 'TextExtractionTool' );

        $tmpName = "var/cache/" . md5( time() ) . '.txt';
        $handle = fopen( $tmpName, "w" );
        fclose( $handle );
        chmod( $tmpName, 0777 );

        exec( "$textExtractionTool $fileName > $tmpName", $ret );

        $metaData = "";
        if ( file_exists( $tmpName ) )
        {
            $fp = fopen( $tmpName, "r" );
            $metaData = fread( $fp, filesize( $tmpName ) );
            fclose( $fp );
            unlink( $tmpName );
        }

        return $metaData;
    }
}

?>


ezinfo.php
<?php
//
// ## BEGIN COPYRIGHT, LICENSE AND WARRANTY NOTICE ##
// SOFTWARE NAME: eZ Tika
// SOFTWARE RELEASE: 1.1
// COPYRIGHT NOTICE: Copyright (C) 2008-2009 Paul Borgermans and eZ Systems AS
// SOFTWARE LICENSE: GNU General Public License v2.0
// NOTICE: >
//   This program is free software; you can redistribute it and/or
//   modify it under the terms of version 2.0  of the GNU General
//   Public License as published by the Free Software Foundation.
//
//   This program is distributed in the hope that it will be useful,
//   but WITHOUT ANY WARRANTY; without even the implied warranty of
//   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
//   GNU General Public License for more details.
//
//   You should have received a copy of version 2.0 of the GNU General
//   Public License along with this program; if not, write to the Free
//   Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,
//   MA 02110-1301, USA.
//
//
// ## END COPYRIGHT, LICENSE AND WARRANTY NOTICE ##
//

/*! \file ezinfo.php
*/

/**
 * eZTikaInfo
 * display meta adn copyright info for eZ Publish eztika extension
 * 
 */

class eZTikaInfo
{
    static function info()
    {
        return array(
            'Name' => "eZ Tika",
            'Version' => "1.2",
            'Copyright' => "Copyright © 2008-2010 Paul Borgermans",
            'Info_url' => "http://projects.ez.no/eztika",
            'License' => "GNU General Public License v2.0",
            '3rdparty_software' =>
                        array ( 'name' => 'Tika',
                                'Version' => '0.6-svn (rev 897576)',
                                'copyright' => 'The Apache Software Foundation.',
                                'license' => 'Apache License, Version 2.0',
                                'info_url' => 'http://lucene.apache.org/tika/' ) );
        }
}

?>
